java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -252,17 +246,14 @@ class EntityHandleManager(
         storageKey: StorageKey,
         schema: Schema
     ): SingletonProxy<R> = proxyMutex.withLock {
-        val storageEndpointProvider: SingletonStorageEndpointProvider<Referencable> =
-            storageEndpointManager.get(
+        singletonStorageProxies.getOrPut(storageKey) {
+            StorageProxy.create(
```

Comment:
Will the lock (proxyMutex) be held until the new proxy registered (`on`) with the service end and returned?

Reply:
Yes, by the time we exit the create method, this will have executed:
https://github.com/PolymerLabs/arcs/pull/6013/files#diff-3da19531e13982f49fdd5a2326a37599R726

Which executes this:
https://github.com/PolymerLabs/arcs/pull/6013/files#diff-d069aacf9d25698e943f9b0a2fb3e7ddR18